### PR TITLE
Fix coverity issues

### PR DESCRIPTION
**Pull request by oliverkurth**
Fixing most issues found by coverity.

Also working around a coverity bug triggered by gcc >= 10.0 - the version isn't parsed correctly. 

Remaining issues:
* 5 instances of `PARSE_ERROR` in `stdlib.h` (we have no control over)
* 7 instances of `PW.BRANCH_PAST_INITIALIZATION` that have no impact
* 2 instances of `PW.INCLUDE_RECURSION` in `libxml` headrers, again not under our control

### CODE REVIEW

File: ci/coverity.sh [hunk 1]
```diff
@@ -16,6 +16,14 @@ CC=cc
 
 cov-configure --config ${COV_CONFIG} --compiler ${CC} --comptype gcc --template
 cov-build --dir ${COV_DIR} --config ${COV_CONFIG} make
+
+# gcc >= 10.0 workaround, see https://www.mail-archive.com/gcc@gcc.gnu.org/msg93814.html
+if grep -qr 'version>0' . --include=*.xml ; then
```
Comment (tapakund) [inline ci/coverity.sh:21]: shouldn't we check for gcc version before doing this? Also, can you share the xml file path.

File: client/remoterepo.c [hunk 2]
```diff
@@ -368,12 +369,19 @@ TDNFCheckRepoMDFileHashFromMetalink(
...
+    {
+        dwError = TDNFChecksumFromHexDigest(hashInfo->value, digest);
+        BAIL_ON_TDNF_ERROR(dwError);
 
+        dwError = TDNFCheckHash(pszFile, digest, hash_Type);
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+    else
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
```
Comment (tapakund) [inline client/remoterepo.c]: should we put more accurate error here?

File: history/history.c [hunk 3]
```diff
@@ -1469,9 +1465,10 @@ int history_update_state(struct history_ctx *ctx, rpmts ts, const char *cmdline)
     history_set_cookie(ctx, cookie);
     ctx->trans_id = trans_id;
 error:
-    if (rc)
+    if (rc) {
         sqlite3_exec(ctx->db, "ROLLBACK;", 0, 0, NULL);
-    else
+        safe_free(current_ids);
+    } else
```
Comment (tapakund) [inline history/history.c:1471]: nit: better to add braces for else as well.

File: tools/cli/main.c [hunk 4]
```diff
@@ -262,19 +262,25 @@ TDNFCliShowVersion(
     PTDNF_CMD_ARGS pCmdArgs
     )
 {
+    uint32_t dwError = 0;
```
Comment (tapakund) [inline tools/cli/main.c:265]: why do we need dwError here at all?

File: tools/cli/main.c [hunk 5]
```diff
@@ -262,19 +262,25 @@ TDNFCliShowVersion(
...
+        CHECK_JD_RC(jd_map_add_string(jd, "Name", TDNFGetPackageName()));
+        CHECK_JD_RC(jd_map_add_string(jd, "Version", TDNFGetVersion()));
         pr_json(jd->buf);
-        jd_destroy(jd);
     }
     else
     {
         pr_info("%s: %s\n", TDNFGetPackageName(), TDNFGetVersion());
     }
+error:
```
Comment (tapakund) [inline tools/cli/main.c:281]: are we doing goto error anywhere from this fn?

**Review by tapakund — COMMENTED** [review]
few comments and queries.

File: ci/coverity.sh (hunk 1 above)
Comment (oliverkurth) [reply ci/coverity.sh:21]: The `if` checks if the version that coverity found starts with `0`, which is the bug when the version is actually `10`.

The xml files are in `${COV_DIR}/emit`, `coverity-intermediate/emit/` under the build directory with our settings in the script.

File: tools/cli/main.c (hunk 4 above)
Comment (oliverkurth) [reply tools/cli/main.c:265]: It's used in the `CHECK_JD_NULL/CHECK_JD_RC` macros.

File: tools/cli/main.c (hunk 5 above)
Comment (oliverkurth) [reply tools/cli/main.c:281]: Yes, again in the macros.

File: client/remoterepo.c (hunk 2 above)
Comment (oliverkurth) [reply client/remoterepo.c]: Changing it to `ERROR_TDNF_INVALID_REPO_FILE` - I think that's better, from looking at the code preceding this.

File: history/history.c (hunk 3 above)
Comment (oliverkurth) [reply history/history.c:1471]: Different coding style in that file.

File: ci/coverity.sh (hunk 1 above)
Comment (tapakund) [reply ci/coverity.sh:21]: instead of directly searching the version in xml file, why don't we do this if the gcc version if greater than or equal to 10?

Comment (oliverkurth) [reply ci/coverity.sh:21]: The `if grep -qr 'version>0' . --include=*.xml` checks for the actual bug. So if coverity fixes the bug we wouldn't do the workaround either. So this is more reliable than just testing for the gcc version.

**Review by tapakund — APPROVED** [review]

**Comment by oliverkurth** [comment]
Thanks @tapakund for reviewing!

**Pull request merged**